homeassistant/components/sensor/netatmo.py

```diff
@@ -0,0 +1,115 @@
+"""
+homeassistant.components.sensor.netatmo
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+This sensor connects to the cloud service provided by netatmo and uses
+the api.
+To successfully connect it needs the client_id and client_secret from
+a netatmo developer account and your username and password from the
+netatmo user account
+"""
+from datetime import timedelta
+import logging
+import requests
+
+from homeassistant.const import (STATE_UNKNOWN)
+from homeassistant.util import Throttle
+from homeassistant.helpers.entity import Entity
+from homeassistant.const import TEMP_CELCIUS
+
+SENSOR_TYPES = {
+    'Temperature': [TEMP_CELCIUS],
+    'Humidity': ['%'],
+    'CO2': ['ppm'],
+    'Noise': ['dB'],
+    'Pressure': ['mb']
+}
+
+sensor_url = "https://api.netatmo.com//api/getstationsdata?access_token="
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'NETAMTO Sensor'
+DEFAULT_METHOD = 'GET'
+
+# Return cached results if last scan was less then this time ago
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=300)
+
+
+# pylint: disable=unused-variable
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Gets a list of the available stations and sets up the sensors. """
+
+    client_id = config.get('client_id', None)
+    client_secret = config.get('client_secret', None)
+    username = config.get('username', None)
+    password = config.get('password', None)
```

Comment:
It seems that those configuration entries are mandatory. Let the setup fail if one is missing with something like that:

```
    if None in (resource, pin):
        _LOGGER.error('Not all required config keys present: %s',
                      ', '.join((CONF_RESOURCE, CONF_PIN)))
        return False
```